### PR TITLE
feat: standalone `TransactionExecutor`

### LINKED ISSUE #14762: Isolate DI for platform-independent infrastructure
We want to reuse the `State`-based initialization currently done [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java#L681) for a "standalone" `TransactionExecutor`.

### LINKED ISSUE #14779: Expose factory to build `TransactionExecutor` from infrastructure and `HederaState`

### LINKED ISSUE #14780: Implement `TransactionExecutor.execute()` via `USER` dispatch

### LINKED ISSUE #14781: Support building contract service `TransactionProcessor` with custom `OperationTracer`s

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Closes #14762 
 - Closes #14779 
 - Closes #14780 
 - Closes #14781 
 - Extracts facility initialization to a `FacilityInitModule` and uses it in a new `ExecutorComponent` to support executing transactions against a `State`, with the only difference from normal execution being that all keys are treated as having signed.
    * See [this test](https://github.com/hashgraph/hedera-services/blob/14762-state-di-module/hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/standalone/TransactionExecutorsTest.java#L102) for how the `TRANSACTION_EXECUTORS` factory can be used to execute an arbitrary `TransactionBody` against a state.

### CODE REVIEW

**Review by IvanKavaldzhiev — APPROVED** [review]
LGTM, tyvm @tinker-michaelj 🥂

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM from `hedera-base`! Thanks @tinker-michaelj

**Review by rbarkerSL — APPROVED** [review]

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/tracers/AddOnEvmActionTracer.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+/**
+ * A {@link OperationTracer} that delegates just the relevant callbacks to a {@link EvmActionTracer}, and all
+ * {@link OperationTracer} callbacks to a list of "add on" tracers.
+ */
+public class AddOnEvmActionTracer implements ActionSidecarContentTracer {
+    private final EvmActionTracer evmActionTracer;
+    private final List<OperationTracer> addOnTracers;
+
+    public AddOnEvmActionTracer(
+            @NonNull final EvmActionTracer evmActionTracer, @NonNull final List<OperationTracer> addOnTracers) {
```
Comment (david-bakin-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/tracers/AddOnEvmActionTracer.java:46]: Looking around here confirms that we don't care about the ordering of the tracers.  Could, if interested, relax the collection type that's being passed around from `List`.  Probably not interested (or even interesting).

**Review by david-bakin-sl — APPROVED** [review]
One small item.  LGTM.

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/ContractServiceComponent.java [hunk 2]
```diff
@@ -21,15 +21,20 @@
...
 @Singleton
 @Component(modules = ContractServiceModule.class)
 public interface ContractServiceComponent {
     @Component.Factory
     interface Factory {
         ContractServiceComponent create(
-                @BindsInstance InstantSource instantSource, @BindsInstance SignatureVerifier signatureVerifier);
+                @BindsInstance InstantSource instantSource,
+                @BindsInstance SignatureVerifier signatureVerifier,
+                @BindsInstance @javax.annotation.Nullable Supplier<List<OperationTracer>> addOnTracers);
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/ContractServiceComponent.java:37]: Dagger2 only changes generated code based on `javax.annotation`s I think...though haven't double-checked in a while. 🤔

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/tracers/AddOnEvmActionTracer.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/tracers/AddOnEvmActionTracer.java:46]: Ah true, could be changed in future PR. 👍

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/ContractServiceComponent.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (david-bakin-sl) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/ContractServiceComponent.java:37]: But see `QueryComponent.create` for another example using `@BindsInstance`.  But there are many other cases where it is used in `@Provides @Singleton` if that counts too.

[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/ContractServiceComponent.java:37]: True...we still use the findbugs annotations some places (probably from force of habit at this point)...but I don't think Dagger2 pays any attention.

**Pull request merged**